WickedEngine/shaders/envMap_skyPS_dynamic.hlsl

```diff
@@ -6,8 +6,16 @@
 
 float4 main(PixelInput input) : SV_TARGET
 {
-	float3 normal = normalize(input.nor);
-	float4 color = float4(GetDynamicSkyColor(normal), 1);
-	color = clamp(color, 0, 65000);
-	return float4(color.rgb,1);
+	float4 color = 0;
+	
+	// When realistic sky 'High Quality' is enabled, let's disable skybox color since we render high quality sky in SkyAtmosphere
+	[branch]
+	if ((GetFrame().options & OPTION_BIT_REALISTIC_SKY_HIGH_QUALITY) == 0)
```

Comment:
Wouldn't it be better to disable the whole pass instead?

Reply:
Yeah my thoughts as well, I'll do a commit later

Reply:
Something to keep in mind, the main color pass is not clearing render target, but it expects the sky drawing will overwrite any untouched pixels. If we don't draw the sky, we will need to clear the render target in that case to not have undefined results show up. This will need to be done in RenderPath3D.cpp in the render pass where DrawSky() is being called. But only clear if sky pixel shader is disabled, otherwise we can avoid the clearing.

Reply:
But we really can't do that for visibility sky, since we allocate tiles and dispatch indirectly?

Reply:
Visibility sky is not a problem, visibility rendering mode will overwrite all pixels guaranteed.

Reply:
Or... maybe it will be a problem now that I think about it. For now, I think you can leave the visibility pipeline though.

Reply:
probably just needs a ClearUav() in the visibility pipeline in this case instead of clear by render pass

Reply:
Perhaps something like this? I didn't do the visibility pipeline, sounds like you had an idea

Reply:
I didn't combine static and gradient background into the atmosphere pass, since I thought that atmosphere should be kept by itself, what do you think?